### DISCUSSION TITLE
Problem with ComfyUI integration (GET /images/api/v1/models HTTP/1.1" 400 Bad Request)

### DISCUSSION THREAD

**Discussion by Gee1111** [open-webui/open-webui#5179] [General]
# Bug Report

## Environment

- **Open WebUI Version:3.16 and 3.18

- **Operating System: linux docker and windows
- **Browser (if applicable): Opera GX and Firefox

![image](https://github.com/user-attachments/assets/e3a40ebb-5ed0-4c69-8660-54a0820a988e)
when i try to configurate my image gen under the admin image settings i get the error ""GET /images/api/v1/models HTTP/1.1" 400 Bad Request" in the logs and "Something went wrong :/ Expecting value: line 1 column 1 (char 0)" in the web ui. My comfyui workflow works perfect

I also tryed with standart ComfyUI workflow - still get this error

The problem persists with comfyUI 0.2 and the latest 0.2.1

Also tryed a fresh install of open webUI

my api json:

{
  "3": {
    "inputs": {
      "text": "",
      "clip": [
        "34",
        1
      ]
    },
    "class_type": "CLIPTextEncode",
    "_meta": {
      "title": "CLIP Text Encode (Prompt) Pos"
    }
  },
  "4": {
    "inputs": {
      "seed": 558740513626235,
      "steps": 20,
      "cfg": 1,
      "sampler_name": "euler",
      "scheduler": "simple",
      "denoise": 1,
      "model": [
        "34",
        0
      ],
      "positive": [
        "8",
        0
      ],
      "negative": [
        "29",
        0
      ],
      "latent_image": [
        "16",
        0
      ]
    },
    "class_type": "KSampler",
    "_meta": {
      "title": "KSampler"
    }
  },
  "8": {
    "inputs": {
      "guidance": 3.5,
      "conditioning": [
        "3",
        0
      ]
    },
    "class_type": "FluxGuidance",
    "_meta": {
      "title": "FluxGuidance"
    }
  },
  "9": {
    "inputs": {
      "samples": [
        "4",
        0
      ],
      "vae": [
        "21",
        0
      ]
    },
    "class_type": "VAEDecode",
    "_meta": {
      "title": "VAE Decode"
    }
  },
  "16": {
    "inputs": {
      "width": 512,
      "height": 512,
      "batch_size": 1
    },
    "class_type": "EmptySD3LatentImage",
    "_meta": {
      "title": "EmptySD3LatentImage"
    }
  },
  "19": {
    "inputs": {
      "unet_name": "flux1-dev-Q8_0.gguf"
    },
    "class_type": "UnetLoaderGGUF",
    "_meta": {
      "title": "Unet Loader (GGUF)"
    }
  },
  "21": {
    "inputs": {
      "vae_name": "ae.safetensors"
    },
    "class_type": "VAELoader",
    "_meta": {
      "title": "Load VAE"
    }
  },
  "27": {
    "inputs": {
      "clip_name1": "t5-v1_1-xxl-encoder-Q8_0.gguf",
      "clip_name2": "clip_l.safetensors",
      "type": "flux"
    },
    "class_type": "DualCLIPLoaderGGUF",
    "_meta": {
      "title": "DualCLIPLoader (GGUF)"
    }
  },
  "28": {
    "inputs": {
      "filename_prefix": "ComfyUI",
      "images": [
        "9",
        0
      ]
    },
    "class_type": "SaveImage",
    "_meta": {
      "title": "Save Image"
    }
  },
  "29": {
    "inputs": {
      "text": "",
      "clip": [
        "34",
        1
      ]
    },
    "class_type": "CLIPTextEncode",
    "_meta": {
      "title": "CLIP Text Encode (Prompt) Neg"
    }
  },
  "34": {
    "inputs": {
      "lora_name": "photorealism.safetensors",
      "strength_model": 1,
      "strength_clip": 1,
      "model": [
        "19",
        0
      ],
      "clip": [
        "27",
        0
      ]
    },
    "class_type": "LoraLoader",
    "_meta": {
      "title": "Load LoRA"
    }
  }
}

**Comment by tkg61** [comment]
have you tried filling in the model node id? i had to do that for mine

**Reply by Gee1111** [reply]
no dosnt work.. i also have no "ckpt_name" cuz im using the gguf model

**Reply by silentoplayz** [reply]
Replace the `ckpt_name` Input Key in the ComfyUI Workflow Node settings within Open WebUI to `unet_name` instead, as that will load your model:
```
"unet_name": "flux1-dev-Q8_0.gguf"
```

Set the Node ID for this to `27`:
```
"27": {
"inputs": {
"clip_name1": "t5-v1_1-xxl-encoder-Q8_0.gguf",
"clip_name2": "clip_l.safetensors",
"type": "flux"
},
```

**Reply by Gee1111** [reply]
lol i already have this in my workflow.. theres nothing to change
i also tested with the default ComfyUI Workflow.. still the same errors..

Is anyone able to coupple the latest versions of Comfy and OWebUI together ?

**Reply by silentoplayz** [reply]
> lol i already have this in my workflow.. theres nothing to change i also tested with the default ComfyUI Workflow.. still the same errors..
> 
> Is anyone able to coupple the latest versions of Comfy and OWebUI together ?

Tested and working on the latest dev branches of Open WebUI and ComfyUI.
![image](https://github.com/user-attachments/assets/14ac88ec-2395-4bc9-8e20-503c7695555e)
![image](https://github.com/user-attachments/assets/cba71011-8429-466d-a821-6916f0caf11f)

**Reply by silentoplayz** [reply]
> Replace the `ckpt_name` Input Key in the ComfyUI Workflow Node settings within Open WebUI to `unet_name` instead, as that will load your model:
> 
> ```
> "unet_name": "flux1-dev-Q8_0.gguf"
> ```
> 
> Set the Node ID for this to `27`:
> 
> ```
> "27": {
> "inputs": {
> "clip_name1": "t5-v1_1-xxl-encoder-Q8_0.gguf",
> "clip_name2": "clip_l.safetensors",
> "type": "flux"
> },
> ```

My sincere apologies. It looks like the Node ID I provided to you initially is incorrect, as I personally overlooked your workflow the first time I attempted to provide help, along with the information I provided. Rather than setting the `unet_name` node ID to `27`, it should be `19` instead, as seen by:
```
"19": {
"inputs": {
"unet_name": "flux1-dev-Q8_0.gguf"
},
```

Your ComfyUI workflow node settings within Open WebUI should look like this for the workflow provided in the [issue](https://github.com/open-webui/open-webui/discussions/5179#discussion-7142025):
![image](https://github.com/user-attachments/assets/a9902393-2d78-4f16-a046-ff1d71e23ab7)

**Reply by Gee1111** [reply]
[flux, lora für OWUI.json](https://github.com/user-attachments/files/16920548/flux.lora.fur.OWUI.json)
i did everything you said.. still get the same error.. can u pls try the flux gguf model with the gguf loaders (for model and dual clip) by urself ?

i added my workflow

the flux-dev Q8 https://huggingface.co/city96/FLUX.1-dev-gguf/tree/main
the T5 encoder Q8 https://huggingface.co/city96/t5-v1_1-xxl-encoder-gguf/tree/main

but i dont think the workflow is the problem cuz with default workflow it didnt work too (even with a fresh install of owebUI on another system)

allways get this error: GET /images/api/v1/models HTTP/1.1" 400 Bad Request

**Reply by silentoplayz** [reply]
> [flux, lora für OWUI.json](https://github.com/user-attachments/files/16920548/flux.lora.fur.OWUI.json) i did everything you said.. still get the same error.. can u pls try the flux gguf model with the gguf loaders (for model and dual clip) by urself ?
> 
> i added my workflow
> 
> the flux-dev Q8 https://huggingface.co/city96/FLUX.1-dev-gguf/tree/main the T5 encoder Q8 https://huggingface.co/city96/t5-v1_1-xxl-encoder-gguf/tree/main

This [flux, lora für OWUI.json](https://github.com/user-attachments/files/16920548/flux.lora.fur.OWUI.json) is not an `API format` workflow exported from ComfyUI. It's just a normal workflow.json export (No worries if you only intend to share this with me and aren't using this as your imported ComfyUI workflow within Open WebUI).

I can't test FLUX, because I have an AMD GPU and using DirectML on Windows 11 for ComfyUI rather than ROCm on Linux.
I also don't know where to find this missing node from the workflow you provided me with. I do have `flux1-schnell-Q5_0.gguf` installed to potentially test out the workflow, but I think DirectML has issues with FLUX in general. I could only get FLUX to generate images within ComfyUI in the earlier days of FLUX's release when a fork of it called `ComfyUI-Zluda` was still hanging around and didn't push an update that simply broke things. I was too lazy to downgrade and test to see if I could be "Comfy" on an outdated version, so I now use ComfyUI with DirectML via Stability Matrix.
![image](https://github.com/user-attachments/assets/2b81e828-3248-445b-bce8-5c2846be39a6)

**Reply by Gee1111** [reply]
yes is know this is the normal format.. but in owebui i imported the right api format..

missing nodes: manager -> install missing custom nodes

**Reply by silentoplayz** [reply]
> missing nodes: manager -> install missing custom nodes

![image](https://github.com/user-attachments/assets/bab8f068-c365-41ef-a47b-512e475e6e52)
![image](https://github.com/user-attachments/assets/9640d0b1-5998-4522-8f20-a0584bc31413)
![image](https://github.com/user-attachments/assets/a6e574f1-ae85-45e1-9e24-cdfaa7aca680)

**Reply by Gee1111** [reply]
look for "ComfyUI-GGUF"  Node ID 54 .. it installs a bunch of custom nodes

**Reply by silentoplayz** [reply]
> look for "ComfyUI-GGUF" Node ID 54 .. it installs a bunch of custom nodes

I genuinely do not see this as a custom node that can be installed within ComfyUI. It says `Channel: dev (Incomplete list)` at the top right, but I'm unsure why that is.
![image](https://github.com/user-attachments/assets/13054e03-d72e-43d4-94b2-bfbb5749ce23)

**Reply by Gee1111** [reply]
ok now it works.. it was a problem with my docker image security configuration..

**Reply by MojoJojo43** [reply]
> ok now it works.. it was a problem with my docker image security configuration..

Can you tell us exactly what you did? Same issue here and I have been grinding away ina ll the wrong places for the last 10 hours lol

**Reply by banguit** [reply]
@Gee1111 Can you tell more what you did? I have similar issue and can't figure this out

**Comment by AIGTWOW** [comment]
May I know what is the docker image security configuration to be set. I have exactly the same problem.
